### PR TITLE
chore: add tests for InfracostComment task

### PR DESCRIPTION
**Pull request by vdmgolub**

### CODE REVIEW

File: tasks/comment/InfracostComment/index.ts [hunk 1]
```diff
@@ -12,12 +11,16 @@ import { promises as fsPromises } from 'fs';
  * @param behavior - post behavior, used by `infracost output` command
  * @return the saved file path
  */
-function generateOutput(path: string, format: string, behavior: string): string {
-  const outputFile = '/tmp/infracost-comment.md';
```
Comment (vdmgolub) [inline tasks/comment/InfracostComment/index.ts:16]: Figured it won't work on windows, guess it's alright to output to the current directory.

File: tasks/comment/InfracostComment/index.ts [hunk 2]
```diff
@@ -29,7 +32,7 @@ function generateOutput(path: string, format: string, behavior: string): string
 
   taskLib.setVariable('INFRACOST_CI_POST_CONDITION', behavior);
 
-  const result = taskLib.execSync(
+  const resultCode = await taskLib.exec(
```
Comment (vdmgolub) [inline tasks/comment/InfracostComment/index.ts:35]: `execSync` can't be mocked with the Azure's library :(

File: tasks/comment/InfracostComment/index.ts [hunk 3]
```diff
@@ -12,12 +11,16 @@ import { promises as fsPromises } from 'fs';
...
+async function generateOutput(path: string, format: string, behavior: string): Promise<string> {
+  const outputFile = 'infracost-comment.md';
 
   let pathList: string[];
   try {
     pathList = JSON.parse(path);
+
+    if (typeof(pathList) === 'string') {
+      pathList = [path];
+    }
```
Comment (vdmgolub) [inline tasks/comment/InfracostComment/index.ts:23]: `"example*.json"` is parsed without errors, falling back to string here, probably there is a better way to do it.

File: tasks/comment/InfracostComment/index.ts [hunk 4]
```diff
@@ -102,22 +108,15 @@ async function run() {
     const tag: string | undefined = taskLib.getInput('tag');
     const dryRun = taskLib.getBoolInput('dryRun');
 
-    const repoProvider = taskLib.getVariable('BUILD_REPOSITORY_PROVIDER');
-
-    if (repoProvider === azureProvider && targetType === 'commit') {
-      const failMessage = 'Azure DevOps Repos only support pull-request targetType';
-      taskLib.setResult(taskLib.TaskResult.Failed, failMessage);
-      return;
-    }
```
Comment (vdmgolub) [inline tasks/comment/InfracostComment/index.ts:111]: Letting compost to handle this.

File: tasks/comment/InfracostComment/tests/mock-helper.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import MockAnswer = require('azure-pipelines-task-lib/mock-answer');
+import MockRun = require('azure-pipelines-task-lib/mock-run');
+import path = require('path');
+
+export default function setupRunner(opts: any): MockRun.TaskMockRunner {
```
Comment (hugorut) [inline tasks/comment/InfracostComment/tests/mock-helper.ts:5]: would prefer having an explicit type here to know the `opts` but not a biggie as it's a test helper.

**Review by hugorut — APPROVED** [review]
Took a little bit of time to wrap my head around this, but looks good

File: tasks/comment/InfracostComment/tests/success.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import MockAnswer = require('azure-pipelines-task-lib/mock-answer');
```
[the remark this answers is not in the document]
Comment (vdmgolub) [reply tasks/comment/InfracostComment/tests/success.ts]: My bad, this was not meant to be committed. Removed the file.

File: tasks/comment/InfracostComment/tests/mock-helper.ts (hunk 5 above)
Comment (vdmgolub) [reply tasks/comment/InfracostComment/tests/mock-helper.ts:5]: Agree, but it's not trivial to extract it from the compost, so I decided to not spend time more time on it 🤷‍♂️.

Comment (hugorut) [reply tasks/comment/InfracostComment/tests/mock-helper.ts:5]: 👍

Comment (vdmgolub) [reply tasks/comment/InfracostComment/tests/mock-helper.ts]: I got what you mean, added an interface 👍 .

**Review by hugorut — APPROVED** [review]

**Pull request merged**